apps/next-evans/lib/dates.ts

```diff
@@ -45,3 +47,18 @@ export function newestFirst<
   }
   throw new Error(`unreachable`);
 }
+
+// hh:mm:ss
+export function formatDuration(durationSeconds: number): string {
```

Comment:
this formatting used to be done on the API, and i think you might have missed some edge cases, like if it's less than an hour, we don't want a leading `0:` or `:`. here are the test tuples from the swift unit tests, can you add a TS unit test to check all of these?

```ts
let cases = [
  [185, `3:05`],
  [11105, `3:05:05`],
  [19183, `5:19:43`],
  [3318, `55:18`],
  [3203, `53:23`],
  [9720, `2:42:00`],
  [3, `3`],
]
```

use the `test.each()` style, rather than manually iterating over the cases.